cep-recipe-serialization.md

```diff
@@ -0,0 +1,1110 @@
+strong_constrains: [list of strong constrains]
+noarch: [list of noarch run exports]
+```
+
+The `run` section also contains a single `run_exports` entry that is used to
+generate the `run_exports` metadata for the package that we are currently
+building. Different from the other `run_exports`, these are "outgoing"
+run_exports.
+
+### Finalized sources
+
+This section contains the finalized sources that were used at build-time. For
+`git` sources, this contains a reference to the commit hash that was used. For
+`url` and `path` sources, this contains the URL and the SHA256 hash of the
+downloaded file.
+
+```yaml
+finalized_sources:
+  - path: ../
+  - git: https://github.com/ros2-gbp/rosidl_typesupport_fastrtps-release.git
+    rev: 7af149d83d7e4a2d4c9ce4ae9682bff4a8ebc28e # here we are storing the commit hash
+    target_directory: ros-humble-fastrtps-cmake-module/src/work
+  - url: https://github.com/watchexec/watchexec/archive/refs/tags/v1.25.1.tar.gz
+    sha256: 9609163c14cd49ec651562838f38b88ed2d370e354af312ddc78c2be76c08d37
+```
+
+The following fields are available for each source:
+
+#### Path source
+
+> [!NOTE] The `sha256` field is added if it is not already present in the source
+> description. (TODO in rattler-build) We should compute the SHA hashes of the
+> patches as well and store it in the source description.
+
+```yaml
+path: # path to the source
+sha256: # computed sha256 hash of the source (this is added if not in the source description)
+patches: # list of patches that should be applied to the source
+target_directory: # optional target directory where the source should be extracted
+file_name: # optional file name
+pub use_gitignore: # optional bool wether to use the .gitignore file when copying a directory
+```
+
+#### Git source
+
+```yaml
+git: # git URL
+rev: # resolved commit hash at build time
+depth: # optional depth to clone the repository
+patches: # list of patches that should be applied to the source
+target_directory: # optional target directory where the source should be extracted
+lfs: # optional bool to request the lfs pull in git source
+```
+
+#### URL source
+
+```yaml
+url: # URL to the source code
+sha256: # sha256 hash of the downloaded file
+md5: # md5 hash of the downloaded file
+file_name: # optional file name to rename the downloaded file
+patches: # list of patches that should be applied to the source
+target_directory: # optional target directory where the source should be extracted
+```
+
+### System tools
+
+This section records all system tools that were used during build execution and
+their versions (if available). This is useful for tracking the tools that were
+used from the host system in order to warn if reproducing the build on a
+different host system might lead to different results. It also records the
+version of `rattler_build` itself, so a future build could be reproduced with
+the same version of `rattler_build`.
+
+```yaml
+system_tools:
+  rattler_build: 0.14.0
```

Comment:
Should it be `rattler-build` to more easily map to the origin package?

Reply:
we made this change :)